app/uk/gov/hmrc/tai/config/ApplicationConfig.scala

```diff
@@ -110,35 +119,42 @@ class ApplicationConfig extends DefaultServicesConfig {
+  lazy val taiRootUri: String = getOptional[String]("microservice.taxPlatformTaiRootUri").getOrElse("")
 
-  def fetchUrl(service: String) =
+  def fetchUrl(service: String): String =
     try {
-      baseUrl(service)
+      servicesConfig.baseUrl(service)
     } catch {
       case ex: RuntimeException => taiRootUri
       case _: Throwable         => s"Unknown Exception: $service-url"
     }
 
-  lazy val isTaiCy3Enabled = configuration.getBoolean("tai.cy3.enabled").getOrElse(false)
-  lazy val numberOfPreviousYearsToShow = configuration.getInt("tai.numberOfPreviousYearsToShow").getOrElse(5)
+  lazy val isTaiCy3Enabled: Boolean = getOptional[Boolean]("tai.cy3.enabled").getOrElse(false)
+  lazy val numberOfPreviousYearsToShow: Int = getOptional[Int]("tai.numberOfPreviousYearsToShow").getOrElse(5)
 }
 
-object ApplicationConfig extends ApplicationConfig
-
-trait FeatureTogglesConfig extends DefaultServicesConfig {
-  val cyPlusOneEnabled = configuration.getBoolean("tai.cyPlusOne.enabled").getOrElse(false)
-  val welshLanguageEnabled = configuration.getBoolean("tai.feature.welshLanguage.enabled").getOrElse(false)
-  val companyCarForceRedirectEnabled =
-    configuration.getBoolean("tai.feature.companyCarForceRedirect.enabled").getOrElse(false)
-  val cyPlus1EstimatedPayEnabled = configuration.getBoolean("tai.cyPlusOne.enabled").getOrElse(false)
-  val webChatEnabled = configuration.getBoolean("tai.webChat.enabled").getOrElse(false)
+trait FeatureTogglesConfig { self: ApplicationConfig =>
+  val cyPlusOneEnabled: Boolean = getOptional[Boolean]("tai.cyPlusOne.enabled").getOrElse(false)
+  val welshLanguageEnabled: Boolean = getOptional[Boolean]("tai.feature.welshLanguage.enabled").getOrElse(false)
+  val companyCarForceRedirectEnabled: Boolean =
+    getOptional[Boolean]("tai.feature.companyCarForceRedirect.enabled").getOrElse(false)
+  val cyPlus1EstimatedPayEnabled: Boolean = getOptional[Boolean]("tai.cyPlusOne.enabled").getOrElse(false)
+  val webChatEnabled: Boolean = getOptional[Boolean]("tai.webChat.enabled").getOrElse(false)
 }
 
-object FeatureTogglesConfig extends FeatureTogglesConfig
+trait AuthConfigProperties { self: ApplicationConfig =>
+
+  val postSignInRedirectUrl: Option[String] = getOptional[String]("microservice.login-callback.url")
+
+  val activatePaperless: Boolean = getOptional[Boolean]("microservice.activatePaperless")
+    .getOrElse(throw new IllegalStateException("Could not find configuration for microservice.activatePaperless"))
+
+  val activatePaperlessEvenIfGatekeeperFails: Boolean =
+    getOptional[Boolean](s"microservice.activatePaperlessEvenIfGatekeeperFails")
+      .getOrElse(throw new IllegalStateException("Could not find configuration for microservice.activatePaperless"))
 
-trait TaiConfig extends DefaultServicesConfig {
-  lazy val baseURL: String = baseUrl("tai")
+  val taxPlatformTaiRootUri: String =
```

Comment:
Are we using line 151, 154 and 158 anywhere inside the code? If not, shall we delete it?

Reply:
Think it would be better to raise a ticket to look at cleaning up the config file, already too many changes on this branch

Reply:
@oc-ste ^?

Reply:
Yeah that's fine, change is mega enough already